aquamarine/src/plumber.rs

```diff
@@ -122,11 +131,36 @@ impl<RT: AquaRuntime, F: ParticleFunctionStatic> Plumber<RT, F> {
         }
 
         // Gather effects and put VMs back
-        let mut effects = vec![];
+        let mut remote_effects = vec![];
+        let mut local_effects: Vec<RoutingEffects> = vec![];
+        let mut interpretation_stats = vec![];
         let mut mailbox_size = 0;
+        let key_manager = self.key_manager.clone();
         for actor in self.actors.values_mut() {
             if let Poll::Ready(result) = actor.poll_completed(cx) {
-                effects.push((result.effects, result.stats));
+                // TODO: filter result.effects and filter out local effects
+
+                interpretation_stats.push(result.stats);
+                let (local_peers, remote_peers): (Vec<_>, Vec<_>) = result
```

Comment:
awkward formatting, what is TODO related to?
```suggestion
                // TODO: filter result.effects and filter out local effects
                interpretation_stats.push(result.stats);
                
                let (local_peers, remote_peers): (Vec<_>, Vec<_>) = result
```

Reply:
It's obsolete